python/src/aac/plugins/gen_design_doc/gen_design_doc_impl.py

```diff
@@ -9,6 +9,7 @@
 from aac.lang.definition_helpers import get_definitions_by_root_key
 from aac.lang.definitions.definition import Definition
 from aac.plugins.plugin_execution import PluginExecutionResult, plugin_result
+from aac.plugins.validators.required_fields import get_required_fields
```

Comment:
Let's move the `get_required_fields` function to somewhere in the `aac.lang.definitions` package if we're going to be using it outside of the validator plugin.

I've been attaching little helpful methods like this to the Definition class, which is in part due to my OOP bias. However, I'd be ok with changing the overall style of those methods to be more functional, I just want the helper functions colocated since they're really similar in scope and goal.

Reply:
For documentation purposes, here's my reasoning for why I moved that function out of the `Definition` class to begin with:

- I'd argue the logical place to put a `get_required_fields` function is in the place where required fields are added to the language, which means (in this case) in the validator plugin that handles required fields.

  Of course, that would mean I'd need to think more about the design because the validator plugin isn't currently responsible for adding/removing required fields so by this logic everything about required fields should be moved to the required fields plugin.

  *QUESTION*: Do we want a future story for doing this?
- If we leave it in the `Definition` class, we introduce a dependency into the core package on a plugin.

  This effectively means that changes to the validator plugin could cause changes to the core `aac` package which doesn't make sense, in my opinion. That kind of thing should go the other way.

  I think we should strive to have the core package follow the open/closed principle where the open part comes from plugins but if a change to a plugin causes the core package to change, that would be an undesirable change. Perhaps this indicates that we should update the design of validator plugins?

Reply:
Per our conversation, I'll resolve this and leave the function in-place for now.